tea_collection/__init__.py

```
@@ -274,3 +563,60 @@ def set_url(self, url: str, sigurl: str):
+        errors = 0
+        errmsg = list()
+        if self.key_exists("url"):
+            if self.format["url"] is None:
+                errors += 1
+                errmsg.append("ERROR: Format has empty URL.")
+        else:
+            errors += 1
+            errmsg.append("ERROR: Format lacks URL.")
+
+        if errors > 0:
+            if self.debug:
```

Ruff

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)